src/exact/ExactMapper.cpp

```diff
@@ -327,13 +327,28 @@ void ExactMapper::map(const Configuration& settings) {
 
 void ExactMapper::coreMappingRoutine(const std::set<unsigned short>& qubitChoice, const CouplingMap& rcm, MappingResults& choiceResults, std::vector<std::vector<std::pair<unsigned short, unsigned short>>>& swaps, long unsigned int limit, unsigned int timeout) {
     const auto& config = results.config;
-
-    // Z3 context
-    context c;
+    using namespace logicbase;
+    // LogicBlock
+    LogicBlock* lb;
```

Comment:
if Z3 is not found, the code is not functional at the moment, right?
Shouldn't this be avoided in order to loosen the dependency on Z3?